pqarrow/builder/optbuilders.go

```diff
@@ -81,6 +81,11 @@ func (b *builderBase) appendValid(n int) {
 	b.length += n
 }
 
+// IsNull returns whether the element at index i is null.
+func (b *builderBase) IsNull(i int) bool {
```

Comment:
I think we should add these utility functions to the optbuilder interface, what do you think?